DelayedClipboard/DelayedClipboardRenderingExplainer.md

```diff
@@ -150,11 +154,72 @@ const clipboardItemInput = new ClipboardItem({['image/png']: blobInput1, ['text/
+
+4. Throw away all the delay rendered formats and put empty data for those formats in the clipboard: It would be a bad user experience, but a web author could also choose to not populate the formats that were registered for delay rendering.
+
+5. Browsers could choose to show a confirmation dialog if there is any delay rendered formats in the clipboard: Although it is similar to option 1, here the text of the dialog explicitly mentions delay rendering of formats which is more informative to the the user. This option has similar concerns as option 1, but it does affect navigations, so it slows down browser/tab shutdown behavior.
+
+More info [here](https://docs.google.com/document/d/1H6ow7RWa4MeycKP3OQBoMhLuMaJct3jSCWW6apcVu9U/edit?usp=sharing)
+
+### Proposal
+
+Proposal is a hybrid of options 2 and 4
+
+1. At least one built-in format must be written to the clipboard without delay as part of the write operation so there will be some data on the clipboard (could be a low fidelity `text/plain` format) if the browser/tab closes and the site doesn't get a chance to generate the data for the delay rendered formats within the timeout window. If the site only writes one format to the clipboard, then it can't be delay rendered.
+
+2. If a delay rendered callback is already running before the page unloads, cancel the callback after a timeout period if the callback hasn't completed yet. When the callback is cancelled, set empty data to the clipboard for that delay rendered format.
+
+3. If site registered for `beforeunload` event to run the callback, run the callback but cancel it if it exceeds a timeout period so it doesn't cause delays in navigation.
+
 ## Privacy and Security Considerations
 
 ### Privacy
 
-No considerable privacy concerns are expected, but we welcome community feedback.
+* Delay rendering of web custom formats
+
+  * Web custom formats are specific to an app ecosystem. When a site registers a callback for a web custom format, it doesn't know where the user is going to paste until the user performs the paste in an app that supports pasting of the web custom format that was copied. On paste, when the web custom format is read from the clipboard, the clipboard calls back into the browser to trigger the callback that was registered for this format so it can return the data to the paste target. The site would know the app or have some fuzzy details about the app where the user is pasting the data into.
```

Comment:
Probably helpful to link at the beginning of this to something where readers can learn what web custom formats are. Either something like https://developer.chrome.com/blog/web-custom-formats-for-the-async-clipboard-api/ or our own explainer if it's up to date.

Reply:
Done.